sync_merkleroots_test.go

```diff
@@ -0,0 +1,104 @@
+package walletclient
+
+import (
+	"context"
+	"testing"
+
+	"github.com/bitcoin-sv/spv-wallet-go-client/fixtures"
+	"github.com/bitcoin-sv/spv-wallet-go-client/models"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSyncMerkleRoots(t *testing.T) {
+
+	t.Run("Should properly sync database when empty", func(t *testing.T) {
+		// setup
+		server := fixtures.MockMerkleRootsAPIResponseNormal()
+		defer server.Close()
+
+		// given
+		repo := fixtures.CreateRepository([]models.MerkleRoot{})
+		client, err := NewWithXPriv(server.URL, fixtures.XPrivString)
+		require.NotNil(t, client.xPriv)
```

Comment:
why we're "testing" the xPriv value here?

Reply:
I modeled it from other tests in this repository. I can remove it but then should we also remove it in other tests?

Reply:
removed